packages/vscode-graphql-syntax/tests/utilities.ts

```diff
@@ -0,0 +1,140 @@
+import * as fs from 'node:fs';
+import { readFile } from 'node:fs/promises';
+import * as path from 'node:path';
+import * as oniguruma from 'vscode-oniguruma';
+import * as tm from 'vscode-textmate';
+
+const ROOT = path.join(__dirname, '..');
+
+type Token = {
+  text: string;
+  scopes: string[];
+};
+
+export async function tokenizeFile(
+  file: string,
+  grammarScopeName: string,
+): Promise<Token[]> {
+  const grammar = await getGrammar(grammarScopeName);
+  const lines = (await readFile(path.join(__dirname, file), 'utf8'))
+    .toString()
+    .split('\n');
+
+  const tokens: Token[] = [];
+
+  let ruleStack = tm.INITIAL;
+  for (const line of lines) {
+    const result = grammar.tokenizeLine(line, ruleStack);
+
+    for (const token of result.tokens) {
+      tokens.push({
+        text: line.slice(token.startIndex, token.endIndex),
+        scopes: token.scopes.filter(scope => scope !== grammarScopeName),
+      });
+    }
+
+    ruleStack = result.ruleStack;
+  }
+
+  return tokens;
+}
+
+const grammarCache: Record<string, tm.IGrammar> = {};
+
+async function getGrammar(scopeName: string) {
+  if (grammarCache[scopeName]) {
+    return grammarCache[scopeName];
+  }
+
+  const grammars = (await loadConfiguration()).map(grammar => {
+    return {
+      grammar,
+      content: fs.readFileSync(grammar.path).toString(),
+    };
+  });
+
+  const grammarMap: { [key: string]: tm.IRawGrammar } = {};
+  const injections: { [scopeName: string]: string[] } = {};
+
+  for (const { grammar, content } of grammars) {
+    const rawGrammar = tm.parseRawGrammar(content, grammar.path);
+    grammarMap[grammar.scopeName || rawGrammar.scopeName] = rawGrammar;
+
+    if (grammar.injectTo) {
+      for (const injectScope of grammar.injectTo) {
+        (injections[injectScope] ||= []).push(grammar.scopeName);
+      }
+    }
+  }
+
+  const registry = new tm.Registry({
+    onigLib: vscodeOnigurumaLib(),
+    loadGrammar: scope => Promise.resolve(grammarMap[scope]),
```

Comment:
does promise here is mandatory? we can do too
```suggestion
    loadGrammar: async scope => grammarMap[scope],
```

Reply:
Yeah, it is

`loadGrammar(scopeName: ScopeName): Promise<IRawGrammar | undefined | null>;`